includes/Jobs/RequestWikiRemoteAIJob.php

```diff
@@ -0,0 +1,395 @@
+				"assistant_id" => $this->config->get( ConfigNames::OpenAIConfig )['assistantid'] ?? ''
+			] );
+
+			$runId = $runData['id'] ?? null;
+
+			$this->logger->debug(
+				'Stage 2 for AI decision of {id}: Message ran.',
+				[
+					'id' => $this->id
+				]
+			);
+
+			$this->logger->debug(
+				'OpenAI returned the following data for stage 2 of {id}: {runData}',
+				[
+					'id' => $this->id,
+					'runData' => json_encode( $runData ),
+				]
+			);
+
+			$this->logger->debug( 'OpenAI returned for stage 2: ' . json_encode( $runData ) );
+
+			if ( !$runId ) {
+				$this->logger->error( 'OpenAI did not return a runId!' );
+				$this->setLastError( 'Run ' . $this->id . ' failed. No runId returned.' );
+				return null;
+			}
+
+			// Step 3: Poll the status of the run
+			$status = 'running';
+			$this->logger->debug( 'Stage 3 for AI decision: Polling status...' );
+
+			while ( $status === 'running' ) {
+				sleep( 5 );
+
+				$this->logger->debug( 'Sleeping for 5 seconds...' );
+				$statusData = $this->createRequest( "/threads/$threadId/runs/$runId" );
+				$status = $statusData['status'] ?? 'failed';
+
+				$this->logger->debug(
+					'Stage 2 for AI decision of {id}: Retrieved run status for {runId}',
+					[
+						'id' => $this->id,
+						'runId' => $runId
+					]
+				);
+
+				$this->logger->debug(
+					'OpenAI returned the following data for stage 3 of {id}: {statusData}',
+					[
+						'id' => $this->id,
+						'statusData' => json_encode( $statusData ),
+					]
+				);
+
+				if ( $status === 'in_progress' ) {
+					$status = 'running';
+				} elseif ( $status === 'failed' ) {
+					$this->logger->error(
+						'Run {runId} failed for {id}! OpenAI returned {statusData}',
+						[
+							'id' => $this->id,
+							'runId' => $runId,
+							'statusData' => json_encode( $statusData )
+						]
+					);
+
+					$this->setLastError( 'Run ' . $runId . ' failed.' );
+
+					return null;
+				}
+			}
+
+			// Step 4: Query for messages in the thread
+			$messagesData = $this->createRequest( "/threads/$threadId/messages" );
+
+			$this->logger->debug(
+				'Stage 4 for AI decision of {id}: Queried for messages in thread {threadId}.',
+				[
+					'id' => $this->id,
+					'threadId' => $threadId
+				]
+			);
+
+			$this->logger->debug(
+				'OpenAI returned the following data for stage 4 of {id}: {messagesData}',
+				[
+					'id' => $this->id,
+					'messagesData' => json_encode( $messagesData ),
+				]
+			);
+
+			$finalResponseContent = $messagesData['data'][0]['content'][0]['text']['value'] ?? '';
+			return json_decode( $finalResponseContent, true );
+		} catch ( Exception $e ) {
+			$this->logger->error( 'HTTP request failed: ' . $e->getMessage() );
+			$this->setLastError( 'An exception occured! The following issue was reported: ' . $e->getMessage() );
+			return null;
+		}
+	}
+
+	private function createRequest( string $endpoint, string $method = 'GET', array $data = [] ): ?array {
```

Comment:
No optional parameters ever in CreateWiki please. Everything should be explicitly passed using named parameters when it is unclear what it does from the variable names you are passing.